File: SmartSalon/SmartSalon/Models/Salon.cs
namespace SmartSalon.Models
{
    public class Salon
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Slug { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public string? Address { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public string? Description { get; set; }
        public string? LogoUrl { get; set; }
        public string ThemeColor { get; set; } = "#1B3A5C";
        public string AdminTheme { get; set; } = "gold";
        public bool IsVip { get; set; } = false;
        public bool IsActive { get; set; } = true;
        public double RatingAvg { get; set; } = 0.0;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // مدیر این سالن کیست؟
        public string ManagerId { get; set; } = string.Empty;
        public ApplicationUser? Manager { get; set; }

        // پرسنل و خدمات این سالن
        public List<Artist> Artists { get; set; } = new();
        public List<SalonService> Services { get; set; } = new();
    }
}

File: smart_salon_app/lib/data/models/salon.dart
import 'artist.dart';
import 'service_item.dart';

class Salon {
  final String id;
  final String name;
  final String? slug;
  final String? phone;
  final String? address;
  final String? description;
  final String? logoUrl;
  final String themeColor;
  final bool isVip;
  final double ratingAvg;
  final List<Artist> artists;
  final List<ServiceItem> services;

  Salon({
    required this.id,
    required this.name,
    this.slug,
    this.phone,
    this.address,
    this.description,
    this.logoUrl,
    this.themeColor = '#1B3A5C',
    this.isVip = false,
    this.ratingAvg = 0.0,
    this.artists = const [],
    this.services = const [],
  });

  factory Salon.fromJson(Map<String, dynamic> json) {
    return Salon(
      id: json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      slug: json['slug'],
      phone: json['phone'],
      address: json['address'],
      description: json['description'],
      logoUrl: json['logoUrl'],
      themeColor: json['themeColor'] ?? '#1B3A5C',
      isVip: json['isVip'] ?? false,
      ratingAvg: (json['ratingAvg'] ?? 0).toDouble(),
      artists: (json['artists'] as List<dynamic>?)
              ?.map((a) => Artist.fromJson(a))
              .toList() ??
          [],
      services: (json['services'] as List<dynamic>?)
              ?.map((s) => ServiceItem.fromJson(s))
              .toList() ??
          [],
    );
  }

  String get initial => name.isNotEmpty ? name[0] : '؟';
}

class SalonListItem {
  final String id;
  final String name;
  final String? logoUrl;
  final double ratingAvg;
  final bool isVip;
  final String? address;
  final int serviceCount;
  final int artistCount;

  SalonListItem({
    required this.id,
    required this.name,
    this.logoUrl,
    this.ratingAvg = 0.0,
    this.isVip = false,
    this.address,
    this.serviceCount = 0,
    this.artistCount = 0,
  });

  factory SalonListItem.fromJson(Map<String, dynamic> json) {
    return SalonListItem(
      id: json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      logoUrl: json['logoUrl'],
      ratingAvg: (json['ratingAvg'] ?? 0).toDouble(),
      isVip: json['isVip'] ?? false,
      address: json['address'],
      serviceCount: json['serviceCount'] ?? 0,
      artistCount: json['artistCount'] ?? 0,
    );
  }

  String get initial => name.isNotEmpty ? name[0] : '؟';
}
